service/raft/raft_group0_client.cc

```diff
@@ -483,6 +483,7 @@ void raft_group0_client::set_query_result(utils::UUID query_id, service::broadca
 
 template group0_command raft_group0_client::prepare_command(schema_change change, group0_guard& guard, std::string_view description);
 template group0_command raft_group0_client::prepare_command(topology_change change, group0_guard& guard, std::string_view description);
+template group0_command raft_group0_client::prepare_command(write_mutations change, group0_guard& guard, std::string_view description);
 template group0_command raft_group0_client::prepare_command(broadcast_table_query change, std::string_view description);
 template group0_command raft_group0_client::prepare_command(write_mutations change, std::string_view description);
```

Comment:
Can this overload be removed? Not that I understand this area.

@kbr-scylla @gleb-cloudius the interface to prepare_command seems awkward. Can it be unified around mutations? After all everything has to become a mutation in order to get persisted.

Reply:
The command does not have to be mutations (broadcast_table_query), but all others are. They are wrapped in the different type to distinguish what state machine they belong to. The alternative is to provide 'cmd_kind' parameter along with mutation array or try to figure it out from tables mutations belong to (not sure it is a good idea).